File: backend/notifications/src/main/java/com/oxygensend/notifications/context/dto/MailDto.java
package com.oxygensend.notifications.context.dto;

import com.oxygensend.notifications.domain.Mail;
import jakarta.validation.constraints.Email;
import jakarta.validation.constraints.NotEmpty;

import java.util.Set;

public record MailDto(@NotEmpty
                      String subject,
                      @NotEmpty
                      String body,
                      Set<@Email String> emails) {

    public Mail toDomain() {
        return new Mail(subject, body);
    }
}

File: backend/notifications/src/main/kotlin/com/oxygensend/notifications/context/dto/MailDto.kt
package com.oxygensend.notifications.context.dto

import jakarta.validation.Valid
import jakarta.validation.constraints.Email
import jakarta.validation.constraints.NotEmpty

data class MailDto(
    val subject: @NotEmpty String,
    val body: @NotEmpty String,
    val emails: @NotEmpty Set<@Valid EmailDto>
) {
    data class EmailDto(
        val address: @Email String,
        val systemId: String? = null
    )
}
